diff --git a/upstream/mmdebstrap/tarfilter b/upstream/mmdebstrap/tarfilter
--- a/upstream/mmdebstrap/tarfilter
+++ b/upstream/mmdebstrap/tarfilter
@@ -273,7 +273,18 @@ def main():
-            if args.strip_components:
-                comps = member.name.split("/")
-                # just as with GNU tar, archive members with less or equal
-                # number of components are not passed through at all
-                if len(comps) <= args.strip_components:
-                    continue
-                member.name = "/".join(comps[args.strip_components :])
+            if args.strip_components:
+                offset = 0
+                stripped = None
+                for _ in range(args.strip_components):
+                    while offset < len(member.name) and member.name[offset] == "/":
+                        offset += 1
+                    if offset == len(member.name):
+                        break
+                    while offset < len(member.name) and member.name[offset] != "/":
+                        offset += 1
+                    if offset == len(member.name):
+                        break
+                    offset += 1
+                else:
+                    stripped = member.name[offset:]
+                if stripped is None:
+                    continue
+                member.name = stripped
