diff --git a/upstream/mmdebstrap/tarfilter b/upstream/mmdebstrap/tarfilter
--- a/upstream/mmdebstrap/tarfilter
+++ b/upstream/mmdebstrap/tarfilter
@@ -49,7 +49,7 @@ class TypeFilterAction(argparse.Action):
     def __call__(self, parser, namespace, values, option_string=None):
         items = getattr(namespace, "typefilter", [])
         match values:
             case "REGTYPE" | "0":
-                items.append(tarfile.REGTYPE)
+                items.extend((tarfile.REGTYPE, tarfile.AREGTYPE))
             case "LNKTYPE" | "1":
                 items.append(tarfile.LNKTYPE)
