diff --git a/upstream/mmdebstrap/coverage.py b/upstream/mmdebstrap/coverage.py
--- a/upstream/mmdebstrap/coverage.py
+++ b/upstream/mmdebstrap/coverage.py
@@ -418,5 +418,6 @@ def main():
         except KeyboardInterrupt:
             proc.terminate()
             proc.wait()
-            break
+            print("interrupted by SIGINT", file=sys.stderr)
+            raise SystemExit(130)
         after = time.time()
