diff --git a/upstream/mmdebstrap/coverage.py b/upstream/mmdebstrap/coverage.py
--- a/upstream/mmdebstrap/coverage.py
+++ b/upstream/mmdebstrap/coverage.py
@@ -415,8 +415,9 @@ def main():
         proc = subprocess.Popen(argv)
         try:
             proc.wait()
         except KeyboardInterrupt:
             proc.terminate()
             proc.wait()
-            break
+            print("interrupted by SIGINT", file=sys.stderr)
+            raise SystemExit(130)
         after = time.time()
