diff --git a/upstream/mmdebstrap/proxysolver b/upstream/mmdebstrap/proxysolver
--- a/upstream/mmdebstrap/proxysolver
+++ b/upstream/mmdebstrap/proxysolver
@@ -15,6 +15,7 @@
 import sys
 import os
 import getpass
+import signal
 
 if not os.path.exists("/usr/lib/apt/solvers/apt"):
     print(
@@ -48,6 +49,13 @@
                 print(line, end="")
                 f.write(line)
             returncode = p.wait()
+    if returncode < 0:
+        sys.stdout.flush()
+        signum = -returncode
+        if signum not in (signal.SIGKILL, signal.SIGSTOP):
+            signal.signal(signum, signal.SIG_DFL)
+            signal.pthread_sigmask(signal.SIG_UNBLOCK, {signum})
+        os.kill(os.getpid(), signum)
     if returncode != 0:
         raise SystemExit(returncode)
 except (FileNotFoundError, PermissionError) as e:
