diff --git a/upstream/mmdebstrap/tarfilter b/upstream/mmdebstrap/tarfilter
--- a/upstream/mmdebstrap/tarfilter
+++ b/upstream/mmdebstrap/tarfilter
@@ -346,9 +346,21 @@ def main():
     ) as out_tar:
+        hardlink_prefix = re.compile(r"^(?:(?:\.\.?/)|/)+")
+        type_excluded_members = set()
         for member in in_tar:
             if path_filter_should_skip(member):
                 continue
             if type_filter_should_skip(member):
+                type_excluded_members.add(hardlink_prefix.sub("", member.name))
                 continue
+            if member.islnk() and hardlink_prefix.sub(
+                "", member.linkname
+            ) in type_excluded_members:
+                print(
+                    "hard-link target excluded by type filter: %s -> %s"
+                    % (member.name, member.linkname),
+                    file=sys.stderr,
+                )
+                exit(1)
             if args.strip_components:
                 comps = member.name.split("/")
                 # just as with GNU tar, archive members with less or equal
